src/fshtypes/FshQuantity.ts

```diff
@@ -10,7 +10,10 @@ export class FshQuantity extends FshEntity {
   toString(): string {
     let str = this.value.toString();
     if (this.unit?.code != null) {
-      str += ` '${this.unit.code}'`;
+      if (this.unit.system) {
+        if (this.unit.system == 'http://unitsofmeasure.org') str += ` '${this.unit.code}'`;
+        else str += ` ${this.unit.system}#${this.unit.code}`;
+      } else str += ` #${this.unit.code}`;
```

Comment:
I mentioned this in the other comment, but I think it might be safer to use the `FshCode.toString` method here if we are not generating a string for the special UCUM units case. If we are, we can keep the handling here since it's a special syntax for a quantity. Otherwise, we're just using regular `FshCode` syntax, so we can take advantage of the logic there.

Reply:
That's a great suggestion! I'm making this change now!